src/core/Classes/Objects/Author.php

```diff
@@ -812,4 +812,45 @@ public static function get_by_id($id)
         return self::get_by_term_id($id);
     }
 
+    /**
+     * Get author posts count with support for post_type.
+     *
+     * @param integer $term_id for the author.
+     * @param string $post_type.
+     *
+     * @return integer $counts
+     */
+    public static function get_author_posts_count($term_id, $post_type = 'post')
+    {
+        global $wpdb;
+        
+        $cache_key = $post_type . '_' . $term_id;
+     
+        $counts = wp_cache_get($cache_key, 'counts');
+
+        if (!$counts) {
+            $counts = $wpdb->get_var(
+                $wpdb->prepare(
+                    "SELECT COUNT(DISTINCT {$wpdb->posts}.ID) FROM {$wpdb->posts}
+                    LEFT JOIN {$wpdb->term_relationships} ON ({$wpdb->posts}.ID = {$wpdb->term_relationships}.object_id)
+                    LEFT JOIN {$wpdb->term_taxonomy} ON ({$wpdb->term_relationships}.term_taxonomy_id = {$wpdb->term_taxonomy}.term_taxonomy_id)
+                    WHERE {$wpdb->term_taxonomy}.taxonomy = %s
+                    AND {$wpdb->term_taxonomy}.term_id = %d
+                    AND {$wpdb->posts}.post_type = %s
+                    AND {$wpdb->posts}.post_status NOT IN ('trash', 'auto-draft')",
+                    ["author", $term_id, $post_type]
+                )
+            );
+            wp_cache_set($cache_key, $counts, 'counts');
```

Comment:
@ojopaul this is working great, but maybe we should give a expire value to the wp_cache_set function? Maybe allowing them to configure the expiration value using a global constant? By default I would set it to expire every one week.

Reply:
Okay, i think you're mistaken `wp_cache_*` to `transient(set_transient)`. Yes, it's possible to set expiry time but unlike transient `wp_cache_` only last during the current page request and only beneficial  if the function run multiple time before page completely load or another function call the method. So, that way, we only make the query once and serve cache for others. 

So, wp_cache_ are not useful on next page load or even on reload and in situation like this, i think using `wp_cache_` is the most appropriate to `transient`. @andergmartins

Reply:
@ojopaul I suggest running the SQL query string through apply_filters() so Permissions or other plugins have a chance to filter it without forcing a redundant query.  The filter should pass $term_id and $post_type as regular arguments, then an optional $args array in case it's ever needed.

Reply:
@agapetry @andergmartins  i agree the need to pass `$term_id` in addition to `$count` and `$post_type` through the filter and i just pushed a commit that added this but  i'm wondering if the `author_posts_count` filter that currently exists shouldn't be enough for third party plugin to filter the count?

My reasons are: 

1.) The current implementation aligned with original WordPress count implementation which pass count via the final count filter and not every part of the query.

2.) In addition, my current implementation was done while putting into consideration the plugin coding style and how similar implementation was done. Wondering if we need to visit similar query that exists on the plugin and pass them through the filter at least, to have code consistency?

I'll make needed changes based on your response to this message.

Reply:
In addition, i changed the filter name to `ppma_author_posts_count` as i believe `author_posts_count` may sound too generic to avoid issue with conflicting other plugin.

Reply:
@ojopaul Regarding the cache, you are right, but that is true only if users are using the native WP cache only. Many users install plugins like Mencached, APC or lightSpeed cache that add persistence to the cached data, according to specific configurations we can't predict. So on those scenarios, maybe the expire time still makes sense.

It makes sense your point about adding another filter to that function, but I see an advantage on what @agapetry proposed. The filter we currently have pass the count number, already calculated. Following Kevin's suggestion we can filter the query itself, making it easier to customize some details without having to rewrite other query.

Btw, you took a good decision renaming that filter 👍🏻

Reply:
@ojopaul The difference between our plugin code and WP Core is that our senior developers are open to my suggestions ;)

Reply:
Noted @agapetry

Reply:
@andergmartins this PR has been updated with query filter and cache expiring days